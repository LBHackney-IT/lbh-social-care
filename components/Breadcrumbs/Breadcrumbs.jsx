import PropTypes from 'prop-types';
import cx from 'classnames';

const Breadcrumbs = ({ label, link, state }) => (
  <li className={cx('govuk-breadcrumbs__list-item')}>
    <a className={`govuk-breadcrumbs__link ${state}`} href={link}>
      {label}
    </a>
  </li>
);

Breadcrumbs.propTypes = {
  label: PropTypes.string.isRequired,
  link: PropTypes.string,
};

export default Breadcrumbs;
