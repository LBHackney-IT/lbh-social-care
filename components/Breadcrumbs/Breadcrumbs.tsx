import Link from 'next/link';
import cx from 'classnames';
import { FormStep } from 'components/Form/types';

import { renderOnCondition } from 'utils/steps';

export interface Props {
  label: string;
  link?: string;
  isCurrentStep: boolean;
}

export interface BreadcrumbProps {
  steps: FormStep[];
  data: Record<string, unknown>;
  path: string;
  currentStepIndex: number;
}

const Breadcrumb = ({
  label,
  link,
  isCurrentStep,
}: Props): React.ReactElement => {
  return (
    <li className="govuk-breadcrumbs__list-item">
      {link ? (
        <Link href={link}>
          <a className="govuk-breadcrumbs__link">{label}</a>
        </Link>
      ) : (
        <span className={cx({ 'govuk-!-font-weight-bold': isCurrentStep })}>
          {label}
        </span>
      )}
    </li>
  );
};

const Breadcrumbs = ({
  steps,
  data,
  path,
  currentStepIndex,
}: BreadcrumbProps): React.ReactElement => {
  return (
    <div className="govuk-breadcrumbs">
      <ol className="govuk-breadcrumbs__list">
        {steps.map((step, index) =>
          renderOnCondition(
            step,
            data,
            <Breadcrumb
              key={step.id}
              label={step.title}
              link={currentStepIndex > index ? `${path}${step.id}` : undefined}
              isCurrentStep={currentStepIndex === index}
            />
          )
        )}
      </ol>
    </div>
  );
};

export default Breadcrumbs;
