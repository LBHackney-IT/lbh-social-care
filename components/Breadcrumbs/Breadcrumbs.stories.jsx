import Breadcrumbs from './Breadcrumbs';

export default {
  title: 'Breadcrumbs',
  component: Breadcrumbs,
};

const Template = (args) => <Breadcrumbs {...args} />;

export const Default = Template.bind({});
Default.args = {
  steps: [
    {
      id: 'first-step',
      title: 'First Step',
    },
    {
      id: 'second-step-not-show',
      title: 'Second Step',
      conditionalRender: ({ hide }) => hide === 'false',
    },
    {
      id: 'second-step',
      title: 'Second Step',
    },
    {
      id: 'third-step',
      title: 'Third Step',
    },
  ],
  path: '/foo/',
  currentStepIndex: 2,
  data: { hide: true },
};
