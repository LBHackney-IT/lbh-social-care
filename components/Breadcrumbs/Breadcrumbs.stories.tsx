import { Story } from '@storybook/react';
import Breadcrumbs, { BreadcrumbProps } from './Breadcrumbs';

export default {
  title: 'Breadcrumbs',
  component: Breadcrumbs,
};

const Template: Story<BreadcrumbProps> = (args) => <Breadcrumbs {...args} />;

export const Default = Template.bind({});
Default.args = {
  steps: [
    {
      id: 'first-step',
      title: 'First Step',
      components: [],
    },
    {
      id: 'second-step-not-show',
      title: 'Second Step',
      conditionalRender: ({ hide }) => hide === 'false',
      components: [],
    },
    {
      id: 'second-step',
      title: 'Second Step',
      components: [],
    },
    {
      id: 'third-step',
      title: 'Third Step',
      components: [],
    },
  ],
  path: '/foo/',
  currentStepIndex: 2,
  data: { hide: true },
};
