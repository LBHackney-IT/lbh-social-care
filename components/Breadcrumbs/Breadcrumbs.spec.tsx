import { render } from '@testing-library/react';

import Breadcrumbs, { BreadcrumbProps } from './Breadcrumbs';

describe('Breadcrumbs component', () => {
  const props: BreadcrumbProps = {
    steps: [
      {
        id: 'first-step',
        title: 'First Step',
        components: [],
      },
      {
        id: 'second-step-not-show',
        title: 'Second Step',
        conditionalRender: ({ hide }: Record<string, unknown>) =>
          hide === 'false',
        components: [],
      },
      {
        id: 'third-step',
        title: 'Third Step',
        components: [],
      },
      {
        id: 'fouth-step',
        title: 'Fourt Step',
        components: [],
      },
    ],
    path: '/foo/',
    currentStepIndex: 2,
    data: { hide: true },
  };
  it('should render properly', () => {
    const { asFragment } = render(<Breadcrumbs {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
