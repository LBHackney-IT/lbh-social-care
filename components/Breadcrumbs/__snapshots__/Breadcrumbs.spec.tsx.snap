// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs component should render properly 1`] = `
<DocumentFragment>
  <div
    class="govuk-breadcrumbs"
  >
    <ol
      class="govuk-breadcrumbs__list"
    >
      <li
        class="govuk-breadcrumbs__list-item"
      >
        <a
          class="govuk-breadcrumbs__link"
          href="/foo/first-step"
        >
          First Step
        </a>
      </li>
      <li
        class="govuk-breadcrumbs__list-item"
      >
        <span
          class="govuk-!-font-weight-bold"
        >
          Third Step
        </span>
      </li>
      <li
        class="govuk-breadcrumbs__list-item"
      >
        <span
          class=""
        >
          Fourt Step
        </span>
      </li>
    </ol>
  </div>
</DocumentFragment>
`;
